File: app/src/main/java/andi/android/madegdk/ui/home/adapter/MovieAdapter.java
package andi.android.madegdk.ui.home.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.RatingBar;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import andi.android.madegdk.R;
import andi.android.madegdk.model.Movie;
import andi.android.madegdk.utils.Util;

public class MovieAdapter extends BaseAdapter {

    private final Context context;
    private ArrayList<Movie> movies;

    public MovieAdapter(Context context) {
        this.context = context;
        movies = new ArrayList<>();
    }

    public void setMovies(ArrayList<Movie> movies) {
        this.movies = movies;
    }

    @Override
    public int getCount() {
        return movies.size();
    }

    @Override
    public Object getItem(int position) {
        return movies.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = LayoutInflater.from(context).inflate(R.layout.item_movie, parent, false);
        }

        ViewHolder viewHolder = new ViewHolder(convertView);
        Movie movie = (Movie) getItem(position);
        viewHolder.bind(movie);
        return convertView;
    }

    private class ViewHolder {
        private TextView titleTV, dateTV, runtimeTV;
        private ImageView posterIV;
        private RatingBar ratingBar;

        ViewHolder(View view) {
            titleTV = view.findViewById(R.id.titleTV);
            dateTV = view.findViewById(R.id.dateTV);
            posterIV = view.findViewById(R.id.posterIV);
            ratingBar = view.findViewById(R.id.ratingBar);
            runtimeTV = view.findViewById(R.id.runtimeTV);
        }

        @SuppressLint("SetTextI18n")
        void bind(Movie movie) {
            titleTV.setText(movie.getTitle());
            dateTV.setText(movie.getDate());
            runtimeTV.setText(movie.getRuntime() + " min");
            Glide.with(context)
                    .load(getDrawableId(movie.getPoster()))
                    .into(posterIV);
            ratingBar.setRating(Util.convertRatingToFloat(movie.getRating()));
        }
    }

    private Integer getDrawableId(String drawableName) {
        return context.getResources().getIdentifier(drawableName, "drawable", context.getPackageName());
    }
}

File: app/src/main/java/andi/android/madegdk/ui/home/adapter/MovieAdapter.kt
package andi.android.madegdk.ui.home.adapter

import andi.android.madegdk.R
import andi.android.madegdk.model.Movie
import andi.android.madegdk.utils.convertRatingToFloat
import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.RatingBar
import android.widget.TextView
import com.bumptech.glide.Glide
import java.util.*

class MovieAdapter(private val context: Context) : BaseAdapter() {
    private var movies: ArrayList<Movie>? = null

    init {
        movies = ArrayList()
    }

    fun setMovies(movies: ArrayList<Movie>) {
        this.movies = movies
    }

    override fun getCount(): Int {
        return movies?.size!!
    }

    override fun getItem(position: Int): Movie? {
        return movies?.get(position)
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var mConvertView = convertView
        if (convertView == null) {
            mConvertView = LayoutInflater.from(context).inflate(R.layout.item_movie, parent, false)

            val viewHolder = ViewHolder(convertView)
            val movie = getItem(position)
            viewHolder.bind(movie!!)
        }
        return mConvertView
    }

    private inner class ViewHolder(view: View?) {
        private val titleTV = view?.findViewById<TextView>(R.id.titleTV)
        private val dateTV = view?.findViewById<TextView>(R.id.dateTV)
        private val runtimeTV = view?.findViewById<TextView>(R.id.runtimeTV)
        private val posterIV = view?.findViewById<ImageView>(R.id.posterIV)
        private val ratingBar = view?.findViewById<RatingBar>(R.id.ratingBar)

        @SuppressLint("SetTextI18n")
        fun bind(movie: Movie) {
            titleTV?.text = "A"
            dateTV?.text = movie.date
            runtimeTV?.text = movie.runtime!!.toString() + " min"
            if (posterIV != null) {
                Glide.with(context)
                        .load(getDrawableId(movie.poster))
                        .into(posterIV)
            }
            ratingBar?.rating = convertRatingToFloat(movie.rating)
        }
    }

    private fun getDrawableId(drawableName: String?): Int {
        return context.resources.getIdentifier(drawableName, "drawable", context.packageName)
    }
}